api/src/main/java/org/bahmni/module/bahmni/ie/apps/model/FormPrivilege.java

```diff
@@ -8,14 +8,45 @@
 import javax.persistence.Entity;
 import javax.persistence.Table;
 import java.io.Serializable;
+import java.util.Date;
 import java.util.List;
 
-public class FormPrivilege extends BaseOpenmrsMetadata {
+public class FormPrivilege extends BaseOpenmrsObject implements java.io.Serializable {
```

Comment:
Why not derive from "BaseOpenmrsData" 
like we are doing in other places? https://github.com/Bahmni/openmrs-module-appointments/blob/master/api/src/main/java/org/openmrs/module/appointments/model/Speciality.java

Reply:
We are also not having any traceability of who created the entry, if you do so.

Reply:
I agree to this point. Will take up this as part of second cut . This needs through investigation as our bahmni.ie.apps hibernate framework has to be changed to accommodate as I am facing issues with lazy initialization, session and bean creation issues when I try to use BaseOpenmrsData and creator.